cmd/state-custom-installer/custom-installer.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+)
+
+const (
+	cmdInstall   = "install"
+	cmdUnInstall = "uninstall"
+)
+
+type Params struct {
+	sourcePath string
+	path       string
+}
+
+func newParams() *Params {
+	return &Params{sourcePath: ".", path: "/tmp"}
+}
+
+func main() {
+	if os.Getenv("VERBOSE") == "true" {
+		logging.CurrentHandler().SetVerbose(true)
+	}
+
+	runErr := run()
+	if runErr != nil {
+		errMsg := errs.Join(runErr, ": ").Error()
+		if locale.IsInputError(runErr) {
+			logging.Debug("state-offline-installer errored out due to input: %s", errMsg)
+		} else {
+			multilog.Critical("state-offline-installer errored out: %s", errMsg)
+		}
+
+		fmt.Fprintln(os.Stderr, errMsg)
+	}
+}
+
+func run() error {
+	args := os.Args
+
+	out, err := output.New("", &output.Config{
+		OutWriter: os.Stdout,
+		ErrWriter: os.Stderr,
+	})
+	if err != nil {
+		return errs.Wrap(err, "Could not initialize outputer")
+	}
+
+	p := primer.New(nil, out, nil, nil, nil, nil, nil, nil, nil, nil)
+
+	params := newParams()
+	cmd := captain.NewCommand(
+		path.Base(os.Args[0]), "", "", p, nil, nil,
+		func(ccmd *captain.Command, args []string) error {
+			out.Print(ccmd.UsageText())
+			return nil
+		},
+	)
+	if isZip(os.Args[0]) {
```

Comment:
We determine if our executable has the backpack attached or not, if not, then we must be the uninstaller only. So, basically, if we have no backpack, then assume we are the standalone uninstaller download. Otherwise, give them both options as usual. Without a backpack we can't install anything anyway.

Reply:
That wasn't my point though. You can get the behaviour you want and address the issue I raised. The issue is that the command registration should not be dependent on what input is passed, instead, it's the execution of the command that should consider the input.

If I run `state-offline-installer --help` it will give me help output for only one or the other command, and the actual execution will behave in ways that the user can't predict based on the help output. That's not the only issue, but a good demonstration of why this is putting the carriage in front of the horse.

Command registration should be independent of input. Command execution may consider input. We should design our commands accordingly.